### PR TITLE
add support for Makefile.PL with and without MANIFEST.SKIP

### PR DESCRIPTION
**Pull request by szabgab**
related to #10

### CODE REVIEW

**Comment by oalders** [comment]
Thanks, @szabgab! Are you able to add a test as well? See the `./t` directory for examples.

**Comment by oalders** [comment]
I'm going to close and re-open to get a CI run going.

**Comment by oalders** [comment]
I've got a PR in place for those linting failures.

**Comment by szabgab** [comment]
ok, once you merge #13 I'll rebase this one and add some tests

**Comment by szabgab** [comment]
@oalders tests were added and they are passing now. Anything else? Shall I squash the commits?

File: t/extutils-makemaker-module-with-manifest.skip.bats [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bats
+
+PATH="$PATH:../../../bin"
+
+setup() {
+  cd t/test-data/extutils-makemaker-module-with-manifest.skip.bats
+}
```
Comment (oalders) [inline t/extutils-makemaker-module-with-manifest.skip.bats:7]: Does `setup()` ever get run?

**Review by oalders — CHANGES_REQUESTED** [review]
Looks great! Thanks for this. Just had a couple of comments. You could squash if you like, or I can squash when I merge.

File: t/extutils-makemaker-module-with-manifest.skip.bats (hunk 1 above)
Comment (szabgab) [reply t/extutils-makemaker-module-with-manifest.skip.bats:7]: no idea, I just copied the `t/module_build_module.bats` file :thinking:

Comment (szabgab) [reply t/extutils-makemaker-module-with-manifest.skip.bats:7]: and as I can see it is the same in `t/dzil_module.bats` as well

Comment (szabgab) [reply t/extutils-makemaker-module-with-manifest.skip.bats:7]: but now I can see I need to include the test files in the GA yaml file as well.

Comment (szabgab) [reply t/extutils-makemaker-module-with-manifest.skip.bats:7]: and it seems the build.pl test is also missing from it.

**Comment by szabgab** [comment]
At this point it is unclear to me why, after the `test-dist` calls `exit 0` the test reports it as having an exit code of 1.
And on my computer using the same version of bats the tests pass.
This needs more investigation.

**Comment by oalders** [comment]
Looks good locally:

```text
bats t
dzil_module.bats
 ✓ dzil cpan-install-build-deps
 ✓ dzil cpan-install-dist-deps
 ✓ dzil build-dist
 ✓ dzil test-dist
 ✓ dzil auto-build-and-test-dist
extutils-makemaker-module-with-manifest.bats
 ✓ Makefile.PL cpan-install-dist-deps
 ✓ Makefile.PL build-dist
 ✓ Makefile.PL test-dist
 - Makefile.PL auto-build-and-test-dist (skipped: Tricky to test under CI)
extutils-makemaker-module-with-manifest.skip.bats
 ✓ Makefile.PL cpan-install-dist-deps
 ✓ Makefile.PL build-dist
 ✓ Makefile.PL test-dist
 - Makefile.PL auto-build-and-test-dist (skipped: Tricky to test under CI)
minil_module.bats
 ✓ minil cpan-install-dist-deps
 ✓ minil build-dist
 ✓ minil test-dist
 - minil auto-build-and-test-dist (skipped: Tricky to test under CI)
module_build_module.bats
 ✓ Build.PL cpan-install-dist-deps
 ✓ Build.PL build-dist
 ✓ Build.PL test-dist
 - Build.PL auto-build-and-test-dist (skipped: Tricky to test under CI)

21 tests, 0 failures, 4 skipped
```

**Comment by oalders** [comment]
Having said that, I think it would make sense to change the workflow to run `bats t` rather than the files individually. That makes it less likely for tests to get missed and the output looks to be very readable.

**Comment by oalders** [comment]
This didn't make sense until I spun up the Docker image and tried to run things locally. I believe this commit is the fix: https://github.com/perl-actions/ci-perl-tester-helpers/pull/15/commits/3fca5a08d018fd33e9a5a499ba9f309c150f1c0c

I've got your commits squashed and in #15

**Comment by szabgab** [comment]
I just tried following the instructions in the `.github/workflows/compile.yml` locally and everything worked and passed (even without your fix copying the file). Really strange. Anyway, then let's close this PR and merge #15
Thank you for looking into it!

**Pull request closed without merging**